packages/indexer/src/indexer.test.ts

```diff
@@ -0,0 +1,221 @@
+import type { StreamDataResponse } from "@apibara/protocol";
+import { type MockBlock, MockClient } from "@apibara/protocol/testing";
+import { describe, expect, it } from "vitest";
+import { run } from "./indexer";
+import { vcr } from "./testing";
+import { type MockRet, mockIndexer } from "./testing/indexer";
+
+describe("Run Test", () => {
+  const client = new MockClient(messages, [{}]);
+
+  it("should stream messages", async () => {
+    const sink = vcr<MockRet>();
+    await run(client, mockIndexer, sink);
+
+    expect(sink.result).toMatchInlineSnapshot(`
+      [
+        {
+          "data": [
+            {
+              "block_number": 5000000,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000000n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000001,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000001n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000002,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000002n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000003,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000003n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000004,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000004n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000005,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000005n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000006,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000006n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000007,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000007n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000008,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000008n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000009,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000009n,
+          },
+        },
+        {
+          "data": [
+            {
+              "block_number": 5000010,
+            },
+          ],
+          "endCursor": {
+            "orderKey": 5000010n,
+          },
+        },
+      ]
+    `);
+  });
+});
+
+const messages: StreamDataResponse<MockBlock>[] = [
```

Comment:
Let's generate the data at runtime :)

```suggestion
const messages: StreamDataResponse<MockBlock>[] = Array.from({ length: 10}, (_, i) => {
  const blockNumber = BigInt(5_000_000 + i);
  return {
    _tag: "data",
    data: {
      finality: "accepted",
      data: [{ block_number: blockNumber }],
      endCursor: { orderKey: blockNumber },
    }
  }
}));
```

Reply:
Maybe need some adjustment but that's the idea.